bundler/lib/bundler/cli/gem.rb

```diff
@@ -213,10 +213,12 @@ def validate_ext_name
     def ask_and_set_test_framework
       test_framework = options[:test] || Bundler.settings["gem.test"]
 
-      if test_framework.nil?
+      if test_framework.to_s.empty?
         Bundler.ui.confirm "Do you want to generate tests with your gem?"
-        result = Bundler.ui.ask "Type 'rspec', 'minitest' or 'test-unit' to generate those test files now and " \
-          "in the future. rspec/minitest/test-unit/(none):"
+        result = Bundler.ui.ask "Type 'rspec', 'minitest' or 'test-unit' to generate those test files now. \n" \
+                                "If Bundler is configured to not generate test files, your choice will only be applied to this instance. \n" \
+                                "Otherwise, future bundle gem calls will use your choice, so -t is not needed if your choice will be the same. \n" \
+                                "This setting can be changed anytime with bundle config gem.test <value>. rspec/minitest/test-unit/(none):"
```

Comment:
I know it's tricky and I might just being annoying, but we have the knowledge about this, so I think we can skip all the extra output when not needed.

So I would check:

* If bundler is configured to not generate test files, only add "Your choice will only be applied to this instance" to the output.

* Otherwise, check whether the `-t` flag has been passed.

  * If `-t` has been passed, add the output "Future bundle gem calls will use your choice, so -t is not needed if your choice will be the same. This setting can be changed anytime with bundle config gem.test <value>."

  * Otherwise, add the output "Future bundle gem calls will use your choice. This setting can be changed anytime with bundle config gem.test <value>.".

This way I think we can educate users about the stuff that's relevant to them, while keeping the output minimal and not too overwhelming.

What do you think?

Reply:
You bring up great points about user facing hints/output. I tweaked the display messages a bit. When `-t` is passed and isn't necessary, I ordered the message last after the gem is created to make it easier to notice. I was debating using `Bundler.ui.warn` but for now am sticking with `Bundler.ui.info`.